sky/provision/kubernetes/utils.py

```diff
@@ -441,10 +482,9 @@ def get_gpu_label_key_value(acc_type: str, check_mode=False) -> Tuple[str, str]:
             for node_name, label_list in node_labels.items():
                 for label, value in label_list:
                     if (label == k8s_acc_label_key and
-                            value == k8s_acc_label_value):
-                        # If a node is found, we can break out of the loop
-                        # and proceed to deploy.
-                        return k8s_acc_label_key, k8s_acc_label_value
+                            label_formatter.get_accelerator_from_label_value(
+                                value).lower() == acc_type.lower()):
+                        return label, value
```

Comment:
nit: `acc_type` and `get_accelerator_from_label_value` are already upper case so .lower() may not be necessary. Also, if we implement `get_label_value` for `GFDLabelFormatter` these changes may not be required.

Reply:
made `get_accelerator_from_label_value` guaranteed to be uppercase, going to remove `lower()` for readability.